transactions/set-approved-id-list/2023/feb-27/add_approved_id_list_and_access_limits.cdc

```diff
@@ -0,0 +1,54 @@
+import FlowIDTableStaking from 0x8624b52f9ddcd04a
+
+/// This transaction adds node IDs to the list of approved nodes in
+/// the ID table. 
+/// It also increases the candidate node limit and slot limits
+/// by the number of nodes who are added
+///
+/// If any of the provided nodes already exist in the ID table, this
+/// transaction will not revert (idempotent)
+
+transaction(newApprovedIDs: [String]) {
+
+    // Local variable for a reference to the ID Table Admin object
+    let adminRef: &FlowIDTableStaking.Admin
+
+    prepare(acct: AuthAccount) {
+        // borrow a reference to the admin object
+        self.adminRef = acct.borrow<&FlowIDTableStaking.Admin>(from: FlowIDTableStaking.StakingAdminStoragePath)
+            ?? panic("Could not borrow reference to staking admin")
+    }
+
+    execute {
+		let existingApprovedIDs = FlowIDTableStaking.getApprovedList()
+			?? panic("Could not load approved list")
+
+		let slotLimits = FlowIDTableStaking.getRoleSlotLimits()
+
+		// add any new node ID which doesn't already exist in the approve list
+		// and increase the candidate node limits and slot limits by 1
+		// for each corresponding node added
+		for newNodeID in newApprovedIDs {
+			if existingApprovedIDs[newNodeID] != nil {
+    			continue
+			}
+
+			let nodeInfo = FlowIDTableStaking.NodeInfo(newNodeID)
+
+			slotLimits[nodeInfo.role] = slotLimits[nodeInfo.role]! + 1
+
+			existingApprovedIDs[newNodeID] = true
+		}
+
+		// set the approved list to the union of existing and new node IDs
+        self.adminRef.setApprovedList(existingApprovedIDs)
+
+		currentAccessSlotLimit = slotLimits[UInt8(5)]
+			?? panic("Could not get access node slot limit")
+
+		slotLimits[UInt8(5)] = currentAccessSlotLimit + 2
```

Comment:
We need atleast 5.
1. One for Bitrue
2. Two for Quick node (they got 3 slots last time and want 2 more)
3. Two for Circle (they want to stake more access nodes as well)

Reply:
can we announce this slot openings in advance (1-2 weeks) to not create misunderstanding in the community. It feels a bit wrong to open like this.

Reply:
Yeah, we're still working out our process for how to open up slots, and this one is definitely a little last minute. We will aim to make sure we open slots with more time to spare and earlier announcement.

This one has to be rushed because it is part of a clean up step for the spork that happened last week, feel free to reach out to the team for more details.